src/memray/_memray.pyx

```diff
@@ -917,6 +923,10 @@ cdef class FileReader:
                     self._memory_snapshots.push_back(reader.getLatestMemorySnapshot())
                 else:
                     break
+
+        if len(self._memory_snapshots) > max_memory_records:
+            self._memory_snapshot_stride = int(ceil(len(self._memory_snapshots) / max_memory_records))
```

Comment:
This isn't working the way we want it to - I think we need to cast one of these to a double. The generated code contains:
```cpp
__pyx_t_9 = PyObject_Length(__pyx_t_6);
...
__pyx_v_self->_memory_snapshot_stride = ((int)ceil((__pyx_t_9 / ((Py_ssize_t)__pyx_v_max_memory_records))));
```
So this is dividing one `size_t` by another (truncating to an integer) and then calling `ceil` on that integer, always returning the same integer.